[Flang][OpenMP] Add translation support for MutexInOutSet and InOutSet

Thirumalai-Shaktivel opened the pull request:
Implementatoin details:
Both Mutexinoutset and Inoutset is recognized as flag=0x4 
and 0x8 respectively, the flags is set to `kmp_depend_info` and 
passed as argument to `__kmpc_omp_task_with_deps` runtime call

kiranchandramohan commented on mlir/test/Target/LLVMIR/openmp-llvm.mlir:
```diff
@@ -2590,6 +2590,34 @@ llvm.func @omp_task_attrs() -> () attributes {
+// CHECK:  store i8 3, ptr %[[dep_type_2]], align 1
+// -----
+// dependence_type: Mutexinoutset
+// CHECK:  %[[dep_arr_addr3:.+]] = alloca [1 x %struct.kmp_dep_info], align 8
```
Nit: convert these check line variables to caps for ease of reading.

Thirumalai-Shaktivel replied:
Sure, I will update it soon.

Thirumalai-Shaktivel replied:
Done

kiranchandramohan reviewed (approved):
LGTM.

Thirumalai-Shaktivel committed:
[Flang][OpenMP] Add translation support for MutexInOutSet and InOutSet
dependence_type in depend clause

Implementatoin details:
Both Mutexinoutset and Inoutset is recognized as flag=0x4
and 0x8 respectively, the flags is set to `kmp_depend_info` and
passed as arg to `__kmpc_omp_task_with_deps` runtime call

Thirumalai-Shaktivel commented:
Thanks for the review!
